engine/src/test/java/com/agiletec/aps/system/services/user/AuthenticationProviderManagerTest.java

```diff
@@ -198,55 +199,103 @@ public void authenticate_1() throws Exception {
+            Assert.fail();
         } catch (UsernameNotFoundException e) {
-            throw e;
-        } finally {
             Mockito.verify(userManager, Mockito.times(1)).getUser(Mockito.anyString(), Mockito.anyString());
             Mockito.verify(userManager, Mockito.times(0)).updateLastAccess(Mockito.any(UserDetails.class));
             Mockito.verify(authorizationManager, Mockito.times(0)).getUserAuthorizations(Mockito.anyString());
             Mockito.verify(tokenManager, Mockito.times(0)).createAccessTokenForLocalUser(Mockito.anyString());
+            throw e;
         }
     }
-    
+
     @Test(expected = AuthenticationServiceException.class)
     public void authenticate_4() throws Exception {
         when(this.userManager.getUser(Mockito.anyString(), Mockito.anyString())).thenThrow(new ApsSystemException("System error"));
         TestingAuthenticationToken authTest = new TestingAuthenticationToken("username", "password");
         try {
             Authentication auth = this.authenticationProviderManager.authenticate(authTest);
+            Assert.fail();
         } catch (UsernameNotFoundException e) {
-            throw e;
-        } finally {
             Mockito.verify(userManager, Mockito.times(1)).getUser(Mockito.anyString(), Mockito.anyString());
             Mockito.verify(userManager, Mockito.times(0)).updateLastAccess(Mockito.any(UserDetails.class));
             Mockito.verify(authorizationManager, Mockito.times(0)).getUserAuthorizations(Mockito.anyString());
             Mockito.verify(tokenManager, Mockito.times(0)).createAccessTokenForLocalUser(Mockito.anyString());
+            throw e;
         }
     }
-    
+
+    @Test
+    public void loadUserByUsername_1() throws Exception {
+        UserDetails activeUser = this.createMockUser("admin", true, false, false);
+        when(this.userManager.getUser(Mockito.anyString())).thenReturn(activeUser);
+        when(this.authorizationManager.getUserAuthorizations("admin")).thenReturn(this.createMockAuthorization());
+        org.springframework.security.core.userdetails.UserDetails userDetails
+                = this.authenticationProviderManager.loadUserByUsername("admin");
+        Assert.assertNotNull(userDetails);
+        Assert.assertEquals("admin", userDetails.getUsername());
+        Assert.assertEquals(2, userDetails.getAuthorities().size());
+        Mockito.verify(userManager, Mockito.times(1)).updateLastAccess(Mockito.any(UserDetails.class));
+        Mockito.verify(userManager, Mockito.times(0)).getUser(Mockito.anyString(), Mockito.anyString());
+        Mockito.verify(userManager, Mockito.times(1)).getUser(Mockito.anyString());
+        Mockito.verify(authorizationManager, Mockito.times(1)).getUserAuthorizations(Mockito.anyString());
+        Mockito.verifyZeroInteractions(this.tokenManager);
+    }
+
+    @Test(expected = UsernameNotFoundException.class)
+    public void loadUserByUsername_2() throws Exception {
```

Comment:
Like test loadUserByUsernamNotFound

Reply:
I used a identical name to the name of the method to test (with a suffix for different cases), but i can change it

Reply:
Ok, lets change it to describe the purpose somewhat

Reply:
Done